XLS-0066d-lending-protocol/README.md

```diff
@@ -0,0 +1,1579 @@
+| `TransactionType` | :heavy_check_mark: | `string`  |   `UINT16`    |   **TODO**    | The transaction type.                    |
+| `LoanID`          | :heavy_check_mark: | `string`  |   `HASH256`   |     `N/A`     | The ID of the Loan object to be updated. |
+| `Flags`           |                    | `string`  |   `UINT32`    |       0       | Specifies the flags for the Loan.        |
+
+##### 3.2.3.1 `Flags`
+
+| Flag Name        |  Flag Value  | Description                                    |
+| ---------------- | :----------: | :--------------------------------------------- |
+| `tfLoanDefault`  | `0x00010000` | Indicates that the Loan should be defaulted.   |
+| `tfLoanImpair`   | `0x00020000` | Indicates that the Loan should be impaired.    |
+| `tfLoanUnimpair` | `0x00040000` | Indicates that the Loan should be un-impaired. |
+
+##### 3.2.3.1 Failure Conditions
+
+- A `Loan` object with the specified `LoanID` does not exist on the ledger.
+- The `Account` submitting the transaction is not the `LoanBroker.Owner`.
+- The `lsfLoanDefault` flag is set on the Loan object. Once a Loan is defaulted, it cannot be modified.
+
+- If `Loan(LoanID).Flags == lsfLoanImpaired` AND `tfLoanImpair` flag is provided.
+
+- `Loan.PaymentRemaining == 0`.
+
+- The `tfDefault` flag is specified and:
+  - `LastClosedLedger.CloseTime` < `Loan.NextPaymentDueDate + Loan.GracePeriod`.
+
+##### 3.2.3.2 State Changes
+
+- If the `tfDefault` flag is specified:
+
+  - Calculate the amount of the Default that First-Loss Capital covers:
+
+    - The default Amount equals the outstanding principal and interest, excluding any funds unclaimed by the Borrower.
+      - `DefaultAmount = (Loan.PrincipalOutstanding + Loan.InterestOutstanding) - Loan.AssetsAvailable`.
+    - Apply the First-Loss Capital to the Default Amount
+      - `DefaultCovered = min((LoanBroker(Loan.LoanBrokerID).DebtTotal x LoanBroker(Loan.LoanBrokerID).CoverRateMinimum)  x LoanBroker(Loan.LoanBrokerID).CoverRateLiquidation, DefaultAmount)`
+    - `DefaultAmount -= DefaultCovered`
+
+  - Update the `Vault` object:
+
+    - Decrease the Total Value of the Vault:
+      - `Vault(LoanBroker(LoanBrokerID).VaultID).AssetsTotal -= DefaultAmount`.
+    - Increase the Asset Available of the Vault by liquidated First-Loss Capital and any unclaimed funds amount:
+      - `Vault(LoanBroker(LoanBrokerID).VaultID).AssetsAvailable += DefaultCovered + Loan.AssetsAvailable`.
+
+- Update the `LoanBroker` object:
+
+  - Decrease the Debt of the LoanBroker:
+    - `LoanBroker(LoanBrokerID).DebtTotal -= `
+    - `Loan.PrincipalOutstanding + Loan.InterestOutstanding + Loan.AssetsAvailable`
+  - Decrease the First-Loss Capital Cover Available:
+    - `LoanBroker(LoanBrokerID).CoverAvailable -= DefaultCovered`
+  - Decrease the number of active Loans:
+    - `LoanBroker(LoanBrokerID).OwnerCount -= 1`
```

Comment:
I don't think we can do this until the loan is deleted. Reasons:
1. It violates the invariant in 3.1.2.3.
2. When the `Loan` is deleted, it will try to decrement the owner count again.
3. The `Loan` still exists.
```suggestion
```